packages/blade/src/components/Tag/Tag.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import { Box } from '~components/Box';
+import type { StyledPropsBlade } from '~components/Box/styledProps';
+import { getStyledProps } from '~components/Box/styledProps';
+import { IconButton } from '~components/Button/IconButton';
+import type { IconComponent } from '~components/Icons';
+import { CloseIcon } from '~components/Icons';
+import { Text } from '~components/Typography';
+import { metaAttribute, MetaConstants } from '~utils/metaAttribute';
+import type { StringChildrenType, TestID } from '~utils/types';
+import { isReactNative } from '~utils';
+import BaseBox from '~components/Box/BaseBox';
+
+type TagProps = {
+  /**
+   * Decides the size of Tag
+   *
+   * @default medium
+   */
+  size?: 'medium' | 'large';
+
+  /**
+   * Leading icon for your Tag
+   */
+  icon?: IconComponent;
+
+  /**
+   * Callback when close icon on Tag is clicked
+   */
+  onDismiss?: ({ value }: { value: StringChildrenType }) => void;
+
+  /**
+   * Text that renders inside Tag
+   */
+  children: StringChildrenType;
+
+  /**
+   * Disable tag
+   */
+  isDisabled?: boolean;
+} & StyledPropsBlade &
+  TestID;
+
+const Tag = ({
+  size = 'medium',
+  icon: Icon,
+  onDismiss,
+  children,
+  isDisabled,
+  testID,
+  ...styledProps
+}: TagProps): React.ReactElement | null => {
+  const [isVisible, setIsVisible] = React.useState(true);
+
+  if (!isVisible) {
```

Comment:
So we handle whether or not to show the UI of tag or null internally. The Tag component itself doesn't really get unmounted, right? If the consumer wanted to show or hide this tag conditionally, there is no way for them to do that? And we have to have faith that the consumer has implemented `onDismiss` handler

Imagine this scenario where we submit `selectedItems` to an API call:
```tsx
const selectedItems = ['Item A', 'Item B', 'Item C']

return selectedItems.map((item) => 
  <Tag 
     onDismiss={()=>removeSlectedItem(item)} // This will remove item from selectedItems array
  >
  {item}
  </Tag>
)
```
Everything is correct here and when the tag is dismissed, the array is changed, but what if the consumer doesn't implement onDismiss?

```tsx
const selectedItems = ['Item A', 'Item B', 'Item C']

return selectedItems.map((item) => 
  <Tag>
  {item}
  </Tag>
)
```
Now, on UI for the end user, selected items will change as and when they dismiss each Item  because we handle that state internally but on code, `selectedItems` array will remain as is and when the API call is made, it will send the full `selectedItems` array which may not be the end-user's intention.


tldr; we shouldn't handle this internally, let consumers handle the visibility of the tag - we can just give them a `onDismiss` callback and they can hide the dismissed item on their end. This will ensure that there will never be a mismatch for the end user. Thoughts?

Reply:
Yupp thanks for pointing out. I actually implemented it this way at the start but then moved to having uncontrolled dismiss to keep it consistent with how we do it in `Alert` right now. In Alert also when we press close icon, we remove Alert and then consumer has to take care of keeping states in sync by calling state update in `onDismiss`

I also saw this is how Ant design and Atlassian design system do as well 
- https://atlassian.design/components/tag/examples#removable
- https://ant.design/components/tag

On the other hand Carbon follows what you're saying
- https://react.carbondesignsystem.com/?path=/story/components-tag--playground&args=filter:!true

I couldn't really find any reasoning for this approach but kept it this way to keep consistency with Ant design and Atlassian's design system.

Would like more opinions on this. Don't really have a strong opinion on either approaches.

Reply:
Its okay on Alert since its just a notification to the user and not necessarily something that will impact the user if the onDismiss is not handled by the developer but on Tags it can have a significant impact